x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/TestCaseSupplier.java

```diff
@@ -214,6 +243,142 @@ public static List<TypedDataSupplier> castToDoubleSuppliersFromRange(Double Min,
         return suppliers;
     }
 
+    public static List<TestCaseSupplier> forBinaryNumericNotCasting(
+        String name,
+        String lhsName,
+        String rhsName,
+        BinaryOperator<Number> expected,
+        DataType expectedType,
+        List<TypedDataSupplier> lhsSuppliers,
+        List<TypedDataSupplier> rhsSuppliers,
+        List<String> warnings,
+        boolean symetric
+    ) {
+        return forBinaryNotCasting(
+            name,
+            lhsName,
+            rhsName,
+            (lhs, rhs) -> expected.apply((Number) lhs, (Number) rhs),
+            expectedType,
+            lhsSuppliers,
+            rhsSuppliers,
+            warnings,
+            symetric
+        );
+    }
+
+    public record StuffForNumericType(Number min, Number max, BinaryOperator<Number> expected, String evaluatorName) {}
+
+    public record AllTheTypeSpecificSettings(StuffForNumericType intStuff, StuffForNumericType longStuff, StuffForNumericType doubleStuff) {
```

Comment:
I dunno better names. But there have to be better names....

Reply:
I dunno, do there? I thought these were pretty descriptive.  That said, I can make them look somewhat more professional.  I hadn't actually intended to leave them with these names, I just forgot.

Reply:
As someone who came up with their online handle when they were 12 I'm not really concerned about professionalism. But, like, my instinct's that words like "stuff" want "what kind of stuff". OTOH, this is inside `TestCaseSupplier` so it's "test case supplier stuff". Dunno. But it'll do.